ToolChain/Assembler/Assembler.py

```diff
@@ -56,79 +50,90 @@ def __init__(self, inputFile=None, outputFile=None):
-        fileContent += b"</AssemblySize>"
-
-        validInternalRef = []   # This will be used to validate for good external reference
-                                # This works because all reference (internal and external) end up
-                                # listed in the internal references anyway.
-
-        # Deal with internal symbols
-        fileContent += b"<InternalSymbols>"
-        for reference in self.parser.referenceDict:
-            if EXPORTED_REFERENCE_INDICATOR not in reference:
-                validInternalRef.append(reference)
-                fileContent += b"<refName>" + reference.encode("utf-8") + b"</refName>"
-                fileContent += b"<refAdd>" + struct.pack(">I", self.parser.referenceDict[reference]) + b"</refAdd>"
-
-                # Update global offsets would it be required
-                # This is required since all global offsets are set to 0 at the moment they are parsed
-                wouldBeExported = EXPORTED_REFERENCE_INDICATOR + " " + reference
-                if wouldBeExported in self.parser.referenceDict.keys():
-                    self.parser.referenceDict[wouldBeExported] = self.parser.referenceDict[reference]
-
-        fileContent += b"</InternalSymbols>"
-
-        # External symbols allow for a file to be linked with another file
-        fileContent += b"<ExternalSymbols>"
-        for reference in self.parser.referenceDict:
-            if EXPORTED_REFERENCE_INDICATOR in reference:
-                isValid = False
-                shortReference = reference.split()[1]
-                for internalReference in validInternalRef:
-                    if shortReference == internalReference:
-                        isValid = True
-                        break
-
-                if isValid:
-                    refName = reference.split()[1]
-                    fileContent += b"<refName>" + refName.encode("utf-8") + b"</refName>"
-                    fileContent += b"<refAdd>" + struct.pack(">I", self.parser.referenceDict[reference]) + b"</refAdd>"
-                else:
-                    raise ValueError("Unresolved reference: {} is exported but "
-                                     "is never declared.".format(reference))
-        fileContent += b"</ExternalSymbols>"
-
-        # Deal with the text section
-        fileContent += b"<Text>"
-        for instruction in self.parser.instructionList:
-            fileContent += instruction
-        fileContent += b"</Text>"
-
-        return fileContent
-
-    def writeAssembledFile(self, file=None, outputData=None):
-        """
-        This will write the assembled file to disk
-        :param file: str, output file path
-        :param outputData: bytes, the data to be written on the disk
+        This method assembles the binary data itself. It will parse each line individually and extract the info needed:
+        1.If the label flag is set to 0, it's a regular instruction, string, number, of memref.
```

Comment:
Part of this comment block should be down in the method around line 95. Method comments should explain what the method does from a "higher level" perspective along with the parameters required by the method. I feel like that level of details, in the current context, should be moved to where the flag is actually set.